src/server/egghead-user-cookies.ts

```diff
@@ -5,23 +5,12 @@ export function setUserCookie(res: NextResponse, user: any) {
   if (user) {
     const {contact_id, email, first_name, last_name, is_pro, is_instructor} =
       user
-    res.cookies.set(
-      EGGHEAD_USER_COOKIE_KEY,
-      JSON.stringify({
-        contact_id,
-        email,
-        first_name,
-        last_name,
-        is_pro,
-        is_instructor,
-      }),
-      {
-        maxAge: 1000 * 60 * 60 * 24 * 2, // 2 days in milliseconds
-        secure: process.env.NODE_ENV === 'production',
-        sameSite: 'strict',
-        domain: process.env.NEXT_PUBLIC_AUTH_COOKIE_DOMAIN,
-      },
-    )
+    res.cookies.set(EGGHEAD_USER_COOKIE_KEY, JSON.stringify(user), {
+      maxAge: 1000 * 60 * 60 * 24 * 2, // 2 days in milliseconds
+      secure: process.env.NODE_ENV === 'production',
+      sameSite: 'strict',
+      domain: process.env.NEXT_PUBLIC_AUTH_COOKIE_DOMAIN,
+    })
```

Comment:
full user object instead of the few attributes. I wonder if just including the user's id would fix most of our pain here?